pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstlTypeConverter.java

```diff
@@ -239,6 +245,12 @@ protected byte[] coerceToBytes(Object value) {
     if (value instanceof byte[]) {
       return (byte[]) value;
     }
+    if (value instanceof ByteBuffer) {
```

Comment:
When do we get ByteBuffer ?

Reply:
It comes from the custom CDC CQL Type (the bigint part [is encoded as ByteBuffer](https://github.com/datastax/cdc-apache-cassandra/blob/cd64bdd03af7a608687b6c54aa614021c62d8027/commons/src/main/java/com/datastax/oss/cdc/CqlLogicalTypes.java#L156)) so I mimicked the same in the tests. The `coerceToBytes` from ByteBuffer will kick in when expression like `value.cqlDecimalField.bigint` are evaluated

Reply:
So this is only for tests ? Can we use instead the type that we'll get from the AVRO deserializer (probably byte[] ?)

Reply:
Not only for testing. If the byte[] was encoded as ByteBuffer, the AVRO deserializer will return ByteBuffer even if the schema type is bytes. 

This how CDC code converts it: https://github.com/datastax/cdc-apache-cassandra/blob/cd64bdd03af7a608687b6c54aa614021c62d8027/commons/src/main/java/com/datastax/oss/cdc/CqlLogicalTypes.java#L146

Reply:
I don't see how the info of which structure was used byte[]/ByteBuffer during serialization would be in the AVRO record...

Reply:
I just realized the conversation from ByteBuffer had been removed from the `coerceToBytes` and is now limited to `coerceToBigInteger` only. However, I'm not sure how the AVRO recognizes ByteBuffer under the hoods.